components/src/motion/bno055/adafruit_bno055.ads

```diff
@@ -0,0 +1,639 @@
+--  met:                                                                    --
+--     1. Redistributions of source code must retain the above copyright    --
+--        notice, this list of conditions and the following disclaimer.     --
+--     2. Redistributions in binary form must reproduce the above copyright --
+--        notice, this list of conditions and the following disclaimer in   --
+--        the documentation and/or other materials provided with the        --
+--        distribution.                                                     --
+--     3. Neither the name of STMicroelectronics nor the names of its       --
+--        contributors may be used to endorse or promote products derived   --
+--        from this software without specific prior written permission.     --
+--                                                                          --
+--   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS    --
+--   "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT      --
+--   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR  --
+--   A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT   --
+--   HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, --
+--   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT       --
+--   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,  --
+--   DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY  --
+--   THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT    --
+--   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE  --
+--   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.   --
+--                                                                          --
+------------------------------------------------------------------------------
+
+--  This packager provides an interface for the Adafruit 9-DOF Absolute
+--  Orientation IMU Fusion Breakout board based on the Bosch BNO055 System in
+--  Package device, integrating a triaxial 14-bit accelerometer, a triaxial
+--  16-bit gyroscope, a triaxial geomagnetic sensor and a 32-bit cortex M0+
+--  microcontroller running Bosch Sensortec sensor fusion software.
+--
+--  See https://www.adafruit.com/products/2472
+--
+--  See the Datasheet: "BNO055 Intelligent 9-axis absolute orientation sensor"
+--  by Bosch Sensortec.
+--  Document number BST-BNO055-DS000-13  (BST_BNO055_DS000_13-838248.pdf)
+--  Version 1.3 or higher, although version 1.2 is likely close enough.
+
+--  The AdaFruit BNO055 breakout board supports either I2C or USART
+--  communication. Therefore, the primary type is abstract because the I/O
+--  facility will be implemented by concrete subclasses.
+
+pragma Restrictions (No_Streams);
+
+with System;
+with Interfaces;    use Interfaces;
+with Ada.Real_Time; use Ada.Real_Time;
```

Comment:
You cannot have a dependency on Ada.Real_Time because this makes the entire components layer unusable for boards that only have ZFP run-time available such as the MicroBit.

I know it's not perfect, but please use HAL.Time.